entity-framework/efcore-and-ef6/porting/port-database.md

```diff
@@ -0,0 +1,39 @@
+1. Validate that the EF Core generated classes are compatible with your code.
+1. For exceptions, either modify the generated classes and update the [model configuration](/ef/core/modeling/) or adapt your code to the model.
+
+Note that although EF Core currently scaffolds everything needed to successfully generate a copy of the database, much of the code is not needed for the database-first approach. A fix for this is tracked in [Issue #10890](/dotnet/efcore/issues/10890). Things that you can safely ignore as not needed include: sequences, constraint names, non-unique indexes and index filters.
+
+## Handling schema changes
+
+When your database is the source of truth, EF Core pulls schema information from the database rather than pushing it via migrations. The typical workflow is to re-run the reverse-engineering step whenever the database schema changes. A comprehensive test suite is valuable for this approach because you can automate the scaffolding process and verify the changes by running your tests.
+
+## Tips for handling model differences
+
+For various reasons, you may want your C# domain model to be shaped differently from the one generated in reverse engineering. In many cases, this means manually updating the code that is auto-generated after every schema change. One way to prevent extra effort when the code is regenerated is to use partial classes for your DbContext and related entities. Then you can keep code related to business logic and properties not tracked in the database in a separate set of class files that won't get overwritten.
+
+ If your model is significantly different from the generated one, but doesn't change frequently, one option to consider is use of the [repository pattern](/dotnet/architecture/microservices/microservice-ddd-cqrs-patterns/infrastructure-persistence-layer-design) as an adapter. The repository can consume the EF Core generated classes and publish the custom classes you use. This may reduce the impact of changes by isolating them to the repository code, rather than having to perform an application-wide refactoring each time the schema changes.
+
+You may want to consider an alternate workflow and follow steps similar to the [hybrid approach](/ef/efcore-and-ef6/porting/port-hybrid). Instead of generating a new set of classes each time, you indicate specific tables to only generate new classes. You keep existing classes "as is" and directly add or remove properties that changed. You then update the model configuration to address any changes in how the database maps to your existing classes.
+
+## Customize the code generation
+
+As of EF Core 6, it is possible to customize the code generated by scaffolding by using [T4 templates](/visualstudio/modeling/code-generation-and-t4-text-templates). See [T4 and EF Core Reverse Engineering](https://www.bricelam.net/2020/02/03/t4-and-efcore.html) for details.
+
```

Comment:
There are also third party tools, that offer additional ways of customizing the generated code. (with link to tools page?)